### PR TITLE
61 modularize memoform

### PR DESCRIPTION
**Pull request by Leepak777**
Now there is a Memoform component that has a map to select location, title and description input.

both createMemo and Dashboard are using this component to create or edit Memo

also just refactored a bit with the map and savedlocations component to remove all the any type

### CODE REVIEW

**Comment by Leepak777** [comment]
just changed the following line to fix bug in #65, it's a small change so I included it here
```scrollWheelZoom={true}```

File: frontend/src/components/CreateMemo/components/LocationMapPicker/Map.tsx [hunk 1]
```diff
@@ -3,26 +3,32 @@ import { useState, useEffect } from "react";
...
-  const [initialPosition, setInitialPosition] = useState([49.27326489299744, -123.10365200042726]);
-  const [position, setPosition] = useState(initialPosition);
+
+const MapForm = (props:{ 
+    selectedLocation:string, 
+    onMapClick:([number,number])=>void
+  }) => {
+
+  const [initialPosition, setInitialPosition] = useState<[number,number]>([49.27326489299744, -123.10365200042726]);
+  const [position, setPosition] = useState<[number,number]>(initialPosition);
```
Comment (jerrydngzh) [inline frontend/src/components/CreateMemo/components/LocationMapPicker/Map.tsx:14]: extract out the default coordinates into a constant, `initialPosition` is sorta redundant here, use that constant instead

File: frontend/src/components/CreateMemo/components/Locations/SavedLocations.tsx [hunk 2]
```diff
@@ -29,7 +34,7 @@ const SavedLocations = ({ id, reloadDropdown, onDropdownReloaded, onLocationSele
 
   useEffect(() => {
     fetchData();
-  }, [id, reloadDropdown, onDropdownReloaded]);
+  }, [currentUser.uid, props.reloadDropdown, props.onDropdownReloaded]);
```
Comment (jerrydngzh) [inline frontend/src/components/CreateMemo/components/Locations/SavedLocations.tsx:37]: why watch on `currentUser.uid` in the useEffect subscriber array?

Comment (Leepak777) [reply frontend/src/components/CreateMemo/components/Locations/SavedLocations.tsx]: oh I think before id was a useState that fetch from query so watch it to fetchData or something, just removed it in new commit

File: frontend/src/components/CreateMemo/components/LocationMapPicker/Map.tsx (hunk 1 above)
Comment (Leepak777) [reply frontend/src/components/CreateMemo/components/LocationMapPicker/Map.tsx]: updated

**Review by jerrydngzh — APPROVED** [review]
lgtlm

**Review by jerrydngzh — APPROVED** [review]
lgtm

**Pull request merged**